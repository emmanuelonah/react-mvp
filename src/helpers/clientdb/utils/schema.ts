/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable class-methods-use-this */
export const BsonType = {
  object: 'object',
  string: 'string',
  init: 'init',
  double: 'double',
};

type StringProperty = {
  bsonType: 'string' | 'string'[];
  description: string;
};

type NumberProperty = {
  description: string;
  minimum: number;
  maximum: number;
};

type ObjectProperty = {
  bsonType: 'object' | 'object'[];
  description: string;
};

interface IntProperty extends NumberProperty {
  bsonType: 'int' | 'int'[];
}

interface DoubleProperty extends NumberProperty {
  bsonType: 'double' | 'double'[];
}

type Properties = StringProperty | IntProperty | DoubleProperty | ObjectProperty;

type SchemaValidator = {
  bsonType: 'object';
  title: string;
  required: string[];
  properties: Record<string, Properties>;
};

export class Schema {
  private _schemas!: Record<string, SchemaValidator>;

  public createSchema(entity: string, schemasValidator: SchemaValidator) {
    this._schemas[entity] = schemasValidator;
  }

  public hasSchemaStructure(entity: string) {
    return this._schemas[entity] != undefined;
  }

  public validatePayload<Payload = Record<string, any>>(
    entity: string,
    payload: Payload
  ): { hasSchemaError: boolean; errorMessage: string } {
    return {} as any;
  }
}
