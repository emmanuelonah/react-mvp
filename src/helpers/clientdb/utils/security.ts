import { __HTTPS__ } from 'utils';

const SECURED_ENV = {
  encrypt() {},
  decrypt() {},
};

const UNSECURED_ENV = {
  encrypt() {},
  decrypt() {},
};

function encrypt() {
  if (__HTTPS__) SECURED_ENV.encrypt();
  else UNSECURED_ENV.encrypt();
}

function decrypt() {
  if (__HTTPS__) SECURED_ENV.decrypt();
  else UNSECURED_ENV.decrypt();
}

export { SECURED_ENV, UNSECURED_ENV, encrypt, decrypt };
