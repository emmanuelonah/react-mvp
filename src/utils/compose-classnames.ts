export function composeClassNames(...classNames: (string | number | null | undefined | boolean)[]) {
  const formattedClassNames = classNames.filter(Boolean).join(' ');
  return formattedClassNames;
}
