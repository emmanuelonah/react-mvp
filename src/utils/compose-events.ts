/* eslint-disable no-unused-vars */
import React from 'react';

type CustomEvent = any | React.SyntheticEvent | Event;

export function composeEvents(ourEvent: (event: CustomEvent) => void, theirEvent?: (event: CustomEvent) => void) {
  return function enclosedEventHandler(event: CustomEvent) {
    if (!event.defaultPrevented) ourEvent(event);

    theirEvent?.(event);
  };
}
