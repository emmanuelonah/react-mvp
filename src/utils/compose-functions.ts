/* eslint-disable no-unused-vars */
export function composeFunctions(...fns: ((...args: any[]) => any)[]) {
  return function applyComposedFunction(...args: any[]) {
    return fns.forEach((fn: (...args: any[]) => any) => fn(...args));
  };
}
