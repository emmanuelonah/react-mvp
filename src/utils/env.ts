const __TEST__ = process.env.NODE_ENV === 'test';
const __DEV__ = process.env.NODE_ENV === 'development';
const __PROD__ = process.env.NODE_ENV === 'production';

export { __TEST__, __DEV__, __PROD__ };
