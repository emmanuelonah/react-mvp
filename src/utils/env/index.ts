const __TEST__ = process.env.NODE_ENV === 'test';
const __DEV__ = process.env.NODE_ENV === 'development';
const __PROD__ = process.env.NODE_ENV === 'production';
const __HTTPS__ = globalThis.location.protocol === 'https:';
const __HTTP__ = globalThis.location.protocol !== 'https:';

export { __TEST__, __DEV__, __PROD__, __HTTPS__, __HTTP__ };
