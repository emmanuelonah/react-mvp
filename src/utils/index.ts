export { __DEV__, __PROD__, __TEST__, __HTTPS__, __HTTP__ } from './env';
export { TestWrapper } from './test';
export { throwError } from './throw-error';
export { composeEvents } from './compose-events';
export { createContext } from './create-context';
export { reportWebVitals } from './report-web-vitals';
export { composeFunctions } from './compose-functions';
export { composeClassNames } from './compose-classnames';
