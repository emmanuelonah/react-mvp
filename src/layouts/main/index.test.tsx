import { render } from '@testing-library/react';

import { TestWrapper } from 'utils';

import { Main } from '.';

describe('<Main/>', () => {
  test('should render Component', () => {
    const { container } = render(
      <TestWrapper>
        <Main>Hello React Testing Library</Main>
      </TestWrapper>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
