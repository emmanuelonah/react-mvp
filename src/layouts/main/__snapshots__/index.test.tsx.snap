// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> should render Component 1`] = `
<main
  class="sc-bcXHqe glvQA-d"
  data-testid="main"
  id="main"
>
  Hello React Testing Library
</main>
`;
